### PR TITLE
Fix Jpms test

### LINKED ISSUE #28724: [Failing Test]: beam_PostCommit_Java_Jpms on Java17 permared
### What happened?

https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Jpms_Direct_Java17/

https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Jpms_Dataflow_Java17/

Introduced in #28291 apparently buildSrc logic for Java versions still need fix

Should not blocking release though - since the released artifact are all built on Java8

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 2 (backlog / disabled test but we think the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #28724

Two issues

- remove redundant options after #28291 - this should fix test on Jenkins

- github action test use Java11 as default, as gradle setting requires it here: https://github.com/apache/beam/blob/170310ee83a11e0f4dd9010068be7ab08f1fc91c/sdks/java/testing/jpms-tests/build.gradle#L24

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
Run Jpms Direct Java 17 PostCommit

**Comment by Abacn** [comment]
Example run on fork: https://github.com/Abacn/beam/actions/runs/6342390089/job/17228019422 the test runs (though failing due to gcs credential not exist in fork). On beam repo currently test jar fails to compile

**Comment by Abacn** [comment]
R: @magicgoody @damccorm

**Comment by Abacn** [comment]
Run Jpms Direct Java 17 PostCommit

**Comment by Abacn** [comment]
Run Jpms Direct Java 17 PostCommit

**Comment by Abacn** [comment]
Run Jpms Direct Java 11 PostCommit

**Comment by Abacn** [comment]
Run Jpms Direct Java 11 PostCommit

**Comment by Abacn** [comment]
Run Jpms Direct Java 17 PostCommit

File: sdks/java/testing/jpms-tests/build.gradle [hunk 1]
```diff
@@ -21,7 +21,14 @@ import groovy.json.JsonOutput
 plugins {
   id 'org.apache.beam.module'
 }
-javaVersion="1.11"
+
+// overwrite javaVersion before applyJavaNature
+if (project.hasProperty("compileAndRunTestsWithJava17")) {
+  javaVersion = '1.17'
+} else {
+  javaVersion = '1.11'
```
Comment (Abacn) [inline sdks/java/testing/jpms-tests/build.gradle:29]: this is subtle. Current Java11/Java17 test are configured differently.

- Java11 passed `-Dorg.gradle.java.home=$JAVA_HOME_11_X64` so all projects (e.g. java core) are also compared with Java11

- Java17 uses `compileAndRunTestsWithJava17` flag so all but this project are compiled with default JDK (Java8)

File: sdks/java/testing/jpms-tests/build.gradle [hunk 2]
```diff
@@ -33,30 +40,15 @@ enableJavaPerformanceTesting()
 description = "Apache Beam :: SDKs :: Java :: Testing :: JPMS Tests"
 ext.summary = "E2E test for Java 9 modules"
 
-// Java 17 needs compileJava to add-exports and add-opens for error prone
-if (project.hasProperty("compileAndRunTestsWithJava17")) {
-  def java17Home = project.findProperty("java17Home")
-  project.tasks.withType(JavaCompile) {
+// direct compileJava to use specified java version.
+project.tasks.compileJava {
```
Comment (Abacn) [inline sdks/java/testing/jpms-tests/build.gradle:44]: This makes both (default JDK is Java9+) and (default JDK is Java8 and pass `java11Home` or `java17Home`) works.

**Comment by damccorm** [comment]
I don't think I understand the current or desired test behavior here well enough to review without doing some more research, so I will defer to @kennknowles for review

File: sdks/java/testing/jpms-tests/build.gradle (hunk 2 above)
Comment (kennknowles) [reply sdks/java/testing/jpms-tests/build.gradle:44]: Seems OK to me if it works. Do you need `setJava11Options`?

**Review by kennknowles — APPROVED** [review]
I _think_ this does what we want. Anyhow I think I have as much confidence after this change as I had before it :-)

**Comment by kennknowles** [comment]
I restarted the failed job, which seems to have failed to upload the cache. Seems there may be concurrency errors in Gradle caching. Not sure if that is GHA related or just Gradle remote cache generally.

File: sdks/java/testing/jpms-tests/build.gradle (hunk 2 above)
Comment (Abacn) [reply sdks/java/testing/jpms-tests/build.gradle:44]: We do not have `setJava11Options` or correspondents in beamModulePlugin

setJava17Options is due to new module export rules needed in Java17 (https://blogs.oracle.com/javamagazine/post/a-peek-into-java-17-continuing-the-drive-to-encapsulate-the-java-runtime-internals), it does the same thing as the current code (removed/simplified in this PR).

File: sdks/java/testing/jpms-tests/build.gradle (hunk 1 above)
Comment (Abacn) [reply sdks/java/testing/jpms-tests/build.gradle:29]: Thanks, noted in https://github.com/apache/beam/issues/28120#issuecomment-1741033265 will be done as part of Java 21 support

**Comment by Abacn** [comment]
thanks, now all tests passed (the one in checks and github action postcommits - commented above)

**Pull request merged**